### ISSUE TITLE
Fast eat (Trying to drop while eating food even the PlayerDropItemEvent getting cancel)

### ISSUE THREAD

**Issue by ItsZodiaX** [pmmp/PocketMine-MP#4936]
### Issue description

- Expected result: Able to not eat food
- Actual result: Able to bypass consume animation and Food got eaten

### Steps to reproduce the issue
1. Hold right-click to eat food
2. Press Q to drop food

### OS and versions

* PocketMine-MP: 4.2.5 (git hash: a6cb3313b05fa103ec0b85767e04a6ad9f693149)
* PHP: 8.0.17
* Using JIT: no
* Server OS: Win11
* Game version: Win10

### Plugins
Plugins (2): DevTools v1.15.0+dev, Test v0.0.1

- If you remove all plugins, does the issue still occur? No
- If the issue is **not** reproducible without plugins:
  - Can you provide sample, *minimal* reproducing code for the issue? If so, paste it in the bottom section
  ```php
  public function onDrop(PlayerDropItemEvent $event) : void{
        $event->cancel();
  }
  ```

### Crashdump, backtrace or other files
https://cdn.discordapp.com/attachments/878545340058468412/959086406486085732/2022-03-31_20-41-07_Trim.mp4

**Comment by dktapps** [comment]
I suppose the reason for this is because `Player->isUsingItem()` is still true due to the event being cancelled. When the client performs a drop, it sends a new item interaction regardless of whether the server actually performed that drop or not.

**Comment by dktapps** [comment]
Related to #4491

**Comment by ItsZodiaX** [comment]
I think https://github.com/pmmp/PocketMine-MP/issues/4491 got fixed already.
I can't eat while dropping anymore but if you cancel PlayerDropItemEvent. It still happens

**Comment by dktapps** [comment]
Yes, I know it's fixed. But this bug has a similar cause to #4491 due to an incomplete solution.
